charts/kong/templates/_helpers.tpl

```diff
@@ -255,12 +255,19 @@ Create KONG_STREAM_LISTEN string
     {{- $listenConfig := dict -}}
     {{- $listenConfig := merge $listenConfig . -}}
     {{- $_ := set $listenConfig "address" "0.0.0.0" -}}
+    {{/* You set NGINX stream listens to UDP using a parameter because reasons. Since our configuration
+         is dual-purpose for both the Service and listen string, we forcibly inject this parameter if
+         that's the Service protocol. The default handles configs that predate the addition of the
+         protocol field, where we only supported TCP. */}}
+    {{- if (eq (default .protocol "TCP") "UDP") -}}
+      {{- $_ := set $listenConfig "parameters" (append .parameters "udp") -}}
+    {{- end -}}
     {{- $unifiedListen = append $unifiedListen (include "kong.singleListen" $listenConfig ) -}}
   {{- end -}}
 
   {{- $listenString := ($unifiedListen | join ", ") -}}
   {{- if eq (len $listenString) 0 -}}
-    {{- $listenString = "off" -}}
+    {{- $listenString = "" -}}
```

Comment:
Q: what's the context here, this never worked?

Reply:
Moved to https://github.com/Kong/charts/pull/379/files#diff-3e8a71242505abb72903a338eef1cdea2cd2c674f877c59378eff807951a1fa8R631-R633

The original version of this handled disabling the listen in the context of a single type of listen (TCP), and the updated version needs to combine multiple `kong.streamListen` results into a single stream listen config. If all are empty, the Kong configuration should be `off`.

Returning the empty string seemed a bit easier to work with elsewhere (in the autoEnv builder). Templating doesn't allow for the most elegant implementations--this is yet another helper that's adding to my increasing list of things that'd be better-handled in Go implementation that provides luxuries like actual types and typed function returns.